api/provision/mock-invocation-fn.ts

```diff
@@ -40,7 +51,66 @@ export async function baseHandler(stateInput: ProvisionRequest): Promise<CspResp
   return cspResponse;
 }
 
-export function createCspResponse(request: ProvisionRequest): CspResponse {
+/**
+ * Make a request to an actual CSP implementation of the ATAT API
+ *
+ * @param request The input provisioning request
+ * @returns the response from the CSP
+ */
+async function makeARealRequest(request: ProvisionRequest): Promise<CspResponse> {
+  const cspConfig = JSON.parse(
+    (await secretsClient.getSecretValue({ SecretId: process.env.CSP_DATA_SECRET! })).SecretString!
+  );
+
+  const baseUrl = cspConfig?.[request.targetCsp.name]?.uri;
+  const url = `${baseUrl}/portfolios`;
+  if (!baseUrl || !baseUrl.startsWith("https://")) {
+    logger.error("Invalid CSP configuration", {
+      jobId: request.jobId,
+      input: {
+        csp: request.targetCsp.name,
+      },
+      configSecretPath: process.env.CSP_DATA_SECRET,
+      retrievedConfig: {
+        cspConfig,
+      },
+    });
+    return {
+      code: 400,
+      content: {
+        details: "Invalid CSP provided",
+      },
+    };
+  }
+
+  const response = await axios.post(url, request.payload, {
+    headers: {
+      Authorization: `Bearer ${(await getToken()).access_token}`,
+      "User-Agent": "ATAT v0.2.0 client",
```

Comment:
We'll eventually want to parameterize the version so we don't lose track and end up confusing ATAT servers but I'm good with a TODO for now

Reply:
Yeah somewhere on my TODO list for awhile has been "Dynamically find version in node project"

Reply:
Ehh but I guess that doesn't help here -- we want the version of the _spec_... hmm

Reply:
Actually... what we probably want to do is write an ATAT client library more generally... And then dynamically get the version of that. Something to think about. Maybe tomorrow when we're talking about tickets we need to write.